File: 67.go
func addBinary(a string, b string) string {
	ans := ""
	lena, lenb := len(a), len(b)
	len := max(lena, lenb)
	cur := 0
	for i := 0; i < len; i++ {
		if i < lena {
			cur += int(a[lena-i-1] - '0')
		}
		if i < lenb {
			cur += int(b[lenb-i-1] - '0')
		}
		ans = strconv.Itoa(cur%2) + ans
		cur /= 2
	}
	if cur > 0 {
		ans = "1" + ans
	}
	return ans
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: 67.java
class Solution {
    public String addBinary(String a, String b) {
        StringBuffer ans = new StringBuffer();
        int lena = a.length(), lenb = b.length();
        int len = Math.max(lena, lenb);
        int cur = 0;
        for(int i = 0;i < len;i ++) {
            if(i < lena) {
                cur += a.charAt(lena- i - 1) - '0';
            }
            if(i < lenb) {
                cur += b.charAt(lenb- i - 1) - '0';
            }
            ans.insert(0, (char)(cur%2 + '0'));
            cur /= 2;
        }
        if(cur > 0) {
            ans.insert(0, '1');
        }
        return ans.toString();
    }
}